packages/site-admin/src/components/sidebar-navigation-item/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import {
+	__experimentalItem as Item,
+	__experimentalHStack as HStack,
+	FlexBlock,
+} from '@wordpress/components';
+import { useContext } from '@wordpress/element';
+import { isRTL } from '@wordpress/i18n';
+import { chevronRightSmall, chevronLeftSmall, Icon } from '@wordpress/icons';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
```

Comment:
Feels like we should copy the router and not import it. We should avoid private APIs.

Reply:
The other thing is that it's not clear to me why we're using the router package at all. Router is generally an "application dependency". In other words, the routing is often something specific:

 - Calypso has its own router...
 - Woo commerce might want embed its own router (which could be the same router as Core).
 - Jetpack might be similar to Woo.

So in other words, we need to think about the context in which these components are going to be used and how to we expect the navigation, route definition ... to work there. 

---

It might mean that we need to remove router ties from the current UI components. Assuming "site-admin" package is about providing UI components.

Reply:
I wasn't assuming the site-admin package was about only providing UI components.
I pictured it as a package that provided all the resources needed to implement the Admin app, where the routing system is a crucial key.

Reply:
It depends on where we see the package being used. If we want to share between Calypso, Jetpack and Woo, we need to decouple from routing because routing is different between Calypso and random wp-admin pages. If it's just for Jetpack and Woo, it may be fine. First, we need to know what contract we're building.

I would say if we want to have both: we need to independent but complementary packages. One for UI, used in all and one to implement a router used by just Jetpack and Woo...